R/hp_datatree.R

```diff
@@ -0,0 +1,141 @@
+dt_root <- function(x) {
+  names(x$Get("level", filterFun = data.tree::isRoot))
+}
+
+dt_root_df <- function(x) {
+  x <- dt_root(x)
+  data.frame(
+    from = x,
+    to = x,
+    stringsAsFactors = FALSE
+  )
+}
+
+dt_collect <- function(x) {
+  requireNamespaceQuietStop("data.tree")
+  source <- data.tree::ToDataFrameNetwork(x)
+  if (length(x$fieldsAll) == 0) {
+    source
+  } else {
+    out <- lapply(x$fieldsAll, function(attribute) {
+      data.tree::ToDataFrameNetwork(x, attribute)[3]
+    })
+
+    vctrs::vec_cbind(source, do.call(vctrs::vec_cbind, out))
+
+  }
+}
+
+#' Create a hierplane object using data.tree
+#'
+#' The `data.tree` package is a popular, general purpose hierarchical data
+#' structure for R. Therefore, `hp_datatree` tries to make it so hierplane is
+#' compatible with `data.tree` objects.
+#'
+#' @param .data A `data.tree` object of class "Node".
+#' @param title A title, defaults to "Hierplane", this serves as the header/title of the hierplane.
+#' @param attributes Attributes to assign to the nodes, these are the annotation in the nodes.
+#' @param link Link connecting each node, theres are the tabs or connections you see between each node.
+#' @param styles Assign styles to hierplane generated from `hierplane_styles()`.
+#' @examples
+#' \dontrun{
+#' library(data.tree)
+#' library(yaml)
+#'
+#' "
+#' name: r4fun
+#' tyler:
+#'   name: Tyler
+#'   job: Data Scientist
+#'   species: Human
+#'   toulouse:
+#'     name: Toulouse
+#'     job: Systems Engineer
+#'     species: Cat
+#'     toulouse:
+#'       name: Toulouse
+#'       job: Systems Engineer
+#'       species: Cat
+#'   ollie:
+#'     name: Ollie
+#'     job: Database Administrator
+#'     species: Dog
+#'   lucas:
+#'     name: Lucas
+#'     job: R Programmer
+#'     species: Rabbit
+#' " -> yaml
+#'
+#' yaml %>%
+#'   yaml.load() %>%
+#'   as.Node() %>%
+#'   hp_datatree(
+#'     title = "r4fun github group",
+#'     link = "species",
+#'     attributes = "job"
+#'   ) %>%
+#'   hierplane(
+#'     theme = "light",
+#'     width = "auto",
+#'     height = "auto"
+#'   )
+#' }
+#' @export
+hp_datatree <- function(.data, title = "Hierplane", attributes = NULL, link = "to",
+                        styles = NULL) {
```

Comment:
any concerns about adding node type as a param?

Reply:
I'll play around with this as well.

Reply:
This is done!

Reply:
```r
library(hierplane)
library(data.tree)
library(yaml)

yaml <- "
name: r4fun
tyler:
  name: Tyler
  job: Data Scientist
  species: Human
  toulouse:
    name: Toulouse
    job: Systems Engineer
    species: Cat
    toulouse:
      name: Toulouse
      job: Systems Engineer
      species: Cat
  ollie:
    name: Ollie
    job: Database Administrator
    species: Dog
  lucas:
    name: Lucas
    job: R Programmer
    species: Rabbit
"

yaml %>% 
  yaml.load() %>% 
  as.Node() %>% 
  hp_datatree(
    node_type = "species", 
    link = "species",
    attributes = "job"
  ) %>% 
  hierplane()
```